packages/docs-browser/src/components/Action/actions/Move.js

```diff
@@ -0,0 +1,134 @@
+import React, {useEffect, useState} from 'react'
+import {connect} from 'react-redux'
+import PropTypes from 'prop-types'
+import {FormattedMessage, injectIntl, intlShape} from 'react-intl'
+import {selection} from 'tocco-util'
+import {Button, LoadMask, StyledButton} from 'tocco-ui'
+import styled from 'styled-components'
+import _isEqual from 'lodash/isEqual'
+
+import DocsBrowser from '../../../main'
+import getNode from '../../../utils/getNode'
+import {close, moveElements} from '../../../modules/move/actions'
+
+const StyledButtonsWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+
+  ${StyledButton} {
+    margin-right: 0;
+  }
+`
+
+const MoveAction = ({selection, onSuccess, onError, intl, context, close, moveElements, isWaiting, isDone}) => {
+  const initialNode = getNode(context.history.location.pathname)
+  const initialParent = {model: 'Docs_list_item', key: `${initialNode.model}/${initialNode.key}`}
+  const [parent, setParent] = useState(initialParent)
+
+  useEffect(() => {
+    if (selection.type === 'QUERY') {
+      onError({
+        message: intl.formatMessage({id: 'client.actions.dms-move.explicitSelection.message'})
+      })
+    }
+  }, [])
+
+  useEffect(() => {
```

Comment:
also this. why not call all the callback directly in the sage instead of setting a state attribute and trigger the callback in the react component. imo a components main concern should be the visualization of data.